src/Sentry/Protocol/Trace.cs

```diff
@@ -50,6 +50,21 @@ internal set
     /// <inheritdoc />
     public bool? IsSampled { get; internal set; }
 
+    private Dictionary<string, object?> _data = new();
```

Comment:
Could potentially back this by a `Lazy<T>` to avoid allocations when no Trace.Data is set.

We try to do this wherever possible - particularly for the trace classes as there can potentially be thousands of these allocated at any one time in larger applications.

Reply:
So lazy is an allocation on its own and there is no computation here.  I can add this.  I could also set the initial size of the dictionary to 0 which is likely better.  Thoughts?

Reply:
We could initialise the dictionary only if it's accessed... something like:
```csharp
Dictionary<string, object?>? _data = null;
Dictionary<string, object?> Data => _data ??= new();
```